controllers/cloud.redhat.com/insightsapp_controller.go

```diff
@@ -73,13 +73,22 @@ func (r *InsightsAppReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, err
 	}
 
-	maker := makers.Maker{
+	maker, err := makers.New(&makers.Maker{
 		App:     &iapp,
 		Base:    &base,
 		Client:  r.Client,
 		Ctx:     ctx,
 		Request: &req,
 		Log:     r.Log,
+	})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if iapp.Spec.ObjectStore != nil {
```

Comment:
Ideally, this will move to some App-specific package.  This is here to show how I intend the new Maker to be used.